### PR TITLE
DOC: examples for Styler.pipe

### PR DESCRIPTION
**Pull request by attack68**
Addresses a recent issue raised in #45686

Screen Shot 2022-02-08 at 19 18 03
Screen Shot 2022-02-08 at 19 18 10

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -3509,41 +3509,85 @@ def pipe(self, func: Callable, *args, **kwargs):
...
+        >>> df.style.format("{:.1f}")
+        ...         .pipe(some_highlights, min_color="green")
+        ...         .highlight_between(left=2, right=5)  # doctest: +SKIP
+
+        .. figure:: ../../_static/style/df_pipe_hl2.png
+
+        **Advanced Use**
+
+        Sometimes it may be necessary to pre-define styling functions, but in the case
+        where those functions rely on the styler, or data, context. Since
```
Comment (mroeschke) [inline pandas/io/formats/style.py:3557]: Nit: `styler, or data, context` -> `styler, data, or context`

File: pandas/io/formats/style.py [hunk 2]
```diff
@@ -3509,41 +3509,85 @@ def pipe(self, func: Callable, *args, **kwargs):
...
-        >>> (df.style
-        ...    .highlight_min(subset=['conversion'], color='yellow')
-        ...    .pipe(format_conversion)
-        ...    .set_caption("Results with minimum conversion highlighted."))
-        ...  # doctest: +SKIP
 
-        .. figure:: ../../_static/style/df_pipe.png
+        **Common Use**
+
+        The most common usage pattern may be to pre-define styling operations which
```
Comment (mroeschke) [inline pandas/io/formats/style.py:3532]: Nit: `The most common usage pattern may be` -> `A common usage pattern is`

**Review by mroeschke — COMMENTED** [review]
Some small comments otherwise LGTM.

Also off topic question, the docstring examples could (in theory) eventually generate different results from the pngs if future changes are made right?

**Comment by attack68** [comment]
> Some small comments otherwise LGTM.
> 
> Also off topic question, the docstring examples could (in theory) eventually generate different results from the pngs if future changes are made right?

yes they could, not sure how to get around this though given this is a visualisation device that is better shown?

**Comment by mroeschke** [comment]
> > Some small comments otherwise LGTM.
> > Also off topic question, the docstring examples could (in theory) eventually generate different results from the pngs if future changes are made right?
> 
> yes they could, not sure how to get around this though given this is a visualisation device that is better shown?

Just curious. Agreed pngs are the better visualization method for these examples.

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
nice @attack68 

should this close the original issue?

**Pull request merged**